packages/gatsby-transformer-remark/src/on-node-create.js

```diff
@@ -17,49 +17,56 @@ module.exports = async function onCreateNode(
   }
 
   const content = await loadNodeContent(node)
-  let data = grayMatter(content, pluginOptions)
 
-  // Convert date objects to string. Otherwise there's type mismatches
-  // during inference as some dates are strings and others date objects.
-  if (data.data) {
-    data.data = _.mapValues(data.data, v => {
-      if (_.isDate(v)) {
-        return v.toJSON()
-      } else {
-        return v
-      }
-    })
-  }
+  try {
+    let data = grayMatter(content, pluginOptions)
+    // Convert date objects to string. Otherwise there's type mismatches
+    // during inference as some dates are strings and others date objects.
+    if (data.data) {
+      data.data = _.mapValues(data.data, v => {
+        if (_.isDate(v)) {
+          return v.toJSON()
+        } else {
+          return v
+        }
+      })
+    }
 
-  const markdownNode = {
-    id: createNodeId(`${node.id} >>> MarkdownRemark`),
-    children: [],
-    parent: node.id,
-    internal: {
-      content: data.content,
-      type: `MarkdownRemark`,
-    },
-  }
+    const markdownNode = {
+      id: createNodeId(`${node.id} >>> MarkdownRemark`),
+      children: [],
+      parent: node.id,
+      internal: {
+        content: data.content,
+        type: `MarkdownRemark`,
+      },
+    }
 
-  markdownNode.frontmatter = {
-    title: ``, // always include a title
-    ...data.data,
-    _PARENT: node.id,
-  }
+    markdownNode.frontmatter = {
+      title: ``, // always include a title
+      ...data.data,
+      _PARENT: node.id,
+    }
 
-  markdownNode.excerpt = data.excerpt
-  markdownNode.rawMarkdownBody = data.content
+    markdownNode.excerpt = data.excerpt
+    markdownNode.rawMarkdownBody = data.content
 
-  // Add path to the markdown file path
-  if (node.internal.type === `File`) {
-    markdownNode.fileAbsolutePath = node.absolutePath
-  }
+    // Add path to the markdown file path
+    if (node.internal.type === `File`) {
+      markdownNode.fileAbsolutePath = node.absolutePath
+    }
 
-  markdownNode.internal.contentDigest = crypto
-    .createHash(`md5`)
-    .update(JSON.stringify(markdownNode))
-    .digest(`hex`)
+    markdownNode.internal.contentDigest = crypto
+      .createHash(`md5`)
+      .update(JSON.stringify(markdownNode))
+      .digest(`hex`)
 
-  createNode(markdownNode)
-  createParentChildLink({ parent: node, child: markdownNode })
+    createNode(markdownNode)
+    createParentChildLink({ parent: node, child: markdownNode })
+  } catch (err) {
+    reporter.panicOnBuild(
+      `Error processing Markdown file ${node.absolutePath}:\n
```

Comment:
I forgot about something earlier - markdown doesn't always come from file, so we will need to check if `node.absolutePath` is set before using it. For example:
```js
`Error processing Markdown ${node.absolutePath ? `file: ${node.absolutePath}` : ``}:\n`
```